Add “API group” to glossary #15906 (#16192)
kubernetes/website: content/en/docs/reference/glossary/api-group.md

---
title: API Group
id: api-group
date: 2019-09-02
full_link: /docs/concepts/overview/kubernetes-api/#api-groups
short_description: >

>   A set of related paths in Kubernetes API.

Reviewer: To me,
```suggestion
  A set of related paths in the Kubernetes API.
```
would be more idiomatic English. What do you think @ashishbharthi ?

>   API group makes it easier to extend the Kubernetes API. The API group is specified in a REST path and in the `apiVersion` field of a serialized object.

Author: ```suggestion
In the Kubernetes API, an API group is a set of related API paths. You can enable or disable each API group by changing the configuration of your API server. You can also disable or enable paths to specific resources.
```

> In the Kubernetes API, an API group is a set of related API paths. You can enable or disable each API group by changing the configuration of your API server. You can also disable or enable paths to specific resources.

Reviewer: - this is quite long for a short description
- the line needs indenting so that this file is valid YAML (it's actually a YAML file that contains Markdown)

Author: @sftim I have updated short description. Let me know what you think.

aka:
tags:
- fundamental
- architecture
---

> API group makes it easier to extend the Kubernetes API. The API group is specified in a REST path and in the `apiVersion` field of a serialized object.

Reviewer: Here's the test that I have in mind for the glossary entry:
Is it possible to take that glossary definition and insert it (via a Hugo shortcode) into https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-groups
so that it makes sense to the reader?

How about adding text along the lines of “In the Kubernetes API, an _API group_ is a set of related API paths. You can enable or disable each API group by changing the configuration of your API server. You can also disable or enable paths to specific resources.”?

Author: ```suggestion
In the Kubernetes API, an API group is a set of related API paths. You can enable or disable each API group by changing the configuration of your API server. You can also disable or enable paths to specific resources.
```

If we add glossary definition to https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-groups, it will be self pointing link since full_link is https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-groups.

<!--more-->

> 

Author: ```suggestion
API group makes it easier to extend the Kubernetes API. The API group is specified in a REST path and in the `apiVersion` field of a serialized object.
```

> * Read [API Group](/docs/concepts/overview/kubernetes-api/#api-groups) for more information

Reviewer: nit:
```suggestion
* Read [API Group](/docs/concepts/overview/kubernetes-api/#api-groups) for more information.
```

> * Read [API Group](/docs/concepts/overview/kubernetes-api/#api-groups) for more information.

Reviewer 2: ```suggestion

* Read [API Group](/docs/concepts/overview/kubernetes-api/#api-groups) for more information.
```

Reviewer 2: This is because of the black friday markdown processor in Hugo
